test/tests/hdgkernels/diffusion/tests

```diff
@@ -0,0 +1,77 @@
+[Tests]
+  design = 'DiffusionHDGKernel.md DiffusionHDGDirichletBC.md DiffusionHDGPrescribedGradientBC.md'
+  issues = '#26406'
+  [diffusion]
+    requirement = 'The system shall be to solve a simple diffusion problem using a hybridized discontinuous Galerkin discretization.'
+    input = 'diffusion.i'
+    csvdiff = 'diffusion_out.csv'
+    type = CSVDiff
+  []
+  [mms]
+    requirement = 'The system shall show second order convergence in the L2 norm of the scalar field with a first order basis using a hybridized discontinuous Galerkin discretization on'
+    [tri]
+      type = PythonUnitTest
+      input = test.py
+      test_case = TestLagrangeTri
+      method = "!dbg"
+      required_python_packages = 'pandas matplotlib'
+      installation_type = in_tree
+      valgrind = 'none'
+      detail = 'simplex elements and'
+    []
+    [quad]
+      type = PythonUnitTest
+      input = test.py
+      test_case = TestLagrangeQuad
+      method = "!dbg"
+      required_python_packages = 'pandas matplotlib'
+      installation_type = in_tree
+      valgrind = 'none'
+      detail = 'tensor-product elements.'
+    []
+  []
+  [rt_mms]
+    requirement = 'The system shall show first order convergence in the L2 norm of the scalar field when using constants for the scalar basis and first order L2 Raviart-Thomas basis functions for the vector field in a hybridized mixed method on'
```

Comment:
do we expect 1st order with RT?

Reply:
Yes the scalar field is approximated with constants, so k + 1 = 1 convergence is what we expect